### PR TITLE
Add api endpoints

### PR DESCRIPTION
**Pull request by tiagomarin**
### :speech_balloon: Feel free to shoot me a message in the pull request if you have any questions.

Thank you for your time!

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -8,4 +12,16 @@
...
+  namespace :api do # /api
+    namespace :v1 do # /api/v1
+      resources :users, only: [:index,:show] do # api/v1/users
+        resources :posts, only: [:index,:show], format: :json do # api/v1/users/:user_id/posts
+          resources :comments, format: :json # api/v1/users/:user_id/posts/:post_id/comments
+        end
+      end
+    end
+  end
+  post "/login", to: "authors#login"
```
Comment (danielufeli) [inline config/routes.rb:26]: - [ ] Nice working on your authentication, kindly check I can't seem to register or login I get the below error in the screenshot when I try.
image

**Review by danielufeli — CHANGES_REQUESTED** [review]
Hi @tiagomarin , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Nice commits messages✔️
- Descriptive PR ✔️
- Good use of GitHub flow ✔️
- Clean Code✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (tiagomarin) [reply config/routes.rb:26]: Hi @danielufeli :wave: 

I used a gem called [doorkeeper](https://github.com/doorkeeper-gem/doorkeeper),

So you have to create an OAuth Application running this in rails console: `Doorkeeper::Application.create(name: "App client", redirect_uri: "", scopes: "")` 

To log in you need to follow [this steps](https://rubyyagi.com/rails-api-authentication-devise-doorkeeper/#setup-doorkeeper-gem:~:text=URL%2DEncoded%20form.-,Login,with%20grant_type%2C%20email%2C%20password%2C%20client_id%20and%20client_secret%20attributes.,-As%20we%20are).
**when you log in, you have to copy the access token**

The way I created the API, to add a comment you need to add `/api/v1` right after the 3000 in the url (as in the image below)
Screenshot from 2022-11-18 23-02-09

and need to use the **access token you copied** (type Bearer) on the Authorization tab (image below)
Screenshot from 2022-11-18 23-09-07

**Review by BenMukebo — COMMENTED** [review]
## STATUS: INVALID ❌ 

Hi  @tiagomarin,

I can't miss felicitating your efforts and energy, you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 

### To improve ⏫ 


- [ ]  Great job on adding linters to your project👏   However, they are not passing yet, Since You have submitted a project, that does have linter errors, I cannot proceed to review your code Kindly fix this issue.  Linter is a tool for identifying and reporting errors when you use standard rules, it is always better to have all linter ch

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@BenMukebo) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit.

**Review by FrederickMih — APPROVED** [review]
Hi @tiagomarin, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Excellent communication with the reviewer :+1: 
- Descriptive PR :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tiagomarin** [comment]
Thanks for reviewing my code guys. 👋

**Review by fobadara — COMMENTED** [review]
Hi @tiagomarin, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Linters are passing
✅ Good job completing the project

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**